### PR TITLE
Crimapp 254 workstreams url changes

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Allows WorkStream values to have different url params.

## Link to relevant ticket
[CRIMAPP-253](https://dsdmoj.atlassian.net/browse/CRIMAPP-253)

## Notes for reviewer
Introduces a WorkStream value object to allow urls to be independent of the work stream names stored in the database and events.

The names of the work streams have changed several times and it seems likely they might change again in the future. The introduction of the WorkStream value object allows presentation of WorkStream names in both text and url to be independent of stored values.

Builds on https://github.com/ministryofjustice/laa-review-criminal-legal-aid/pull/480 (which should be merged before these changes).

## How to manually test the feature

[CRIMAPP-253]: https://dsdmoj.atlassian.net/browse/CRIMAPP-253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -13,5 +13,5 @@ feature_flags:
     production: false
   work_stream:
     local: true
-    staging: true
+    staging: false
```
Comment (hiboabd) [inline config/settings.yml:16]: Has this been accidentally changed?

File: app/controllers/concerns/work_streamable.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    raise Allocating::WorkStreamNotFound unless current_user.work_streams.include?(work_stream)
+
+    session[:current_work_stream] = work_stream_param
+
+    @current_work_stream = work_stream
+  end
+
+  def current_work_stream
+    # TODO: enable when viewing applications by work stream feature is live
+    raise 'Errorrrrr' unless FeatureFlags.work_stream.enabled?
```
Comment (hiboabd) [inline app/controllers/concerns/work_streamable.rb:23]: If this module is only include if the work stream ff is enabled then is this necessary?

File: app/views/casework/crime_applications/index.html.erb [hunk 3]
```diff
@@ -7,10 +7,10 @@
 <% if FeatureFlags.work_stream.enabled? %>
   <div class="govuk-tabs">
     <ul class="govuk-tabs__list" role="tablist">
-      <% CaseworkerPresenter::CASEWORKER_COMPETENCIES.each_with_index do |competency| %>
+      <% current_user.work_streams.each do |work_stream| %>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/index.html.erb:10]: 👍

File: config/locales/en/casework.yml [hunk 4]
```diff
@@ -65,6 +65,7 @@ en:
     codefendants: Co-defendants
     correspondence_address: Correspondence address
     criminal_applications_team: CAT 1
+    criminal_applications_team_2: CAT 2
```
Comment (hiboabd) [inline config/locales/en/casework.yml:68]: Does the national crime team locale need removing too?

File: spec/shared_contexts/with_a_logged_in_user.rb [hunk 5]
```diff
@@ -1,6 +1,12 @@
 RSpec.shared_context 'with a logged in user', shared_context: :metadata do
   before do
     current_user
+
+    # Supervisors get all competencies by default
```
Comment (hiboabd) [inline spec/shared_contexts/with_a_logged_in_user.rb:5]: 👍

File: spec/shared_contexts/with_a_logged_in_user.rb [hunk 6]
```diff
@@ -1,6 +1,12 @@
...
+    unless current_user.supervisor?
```
Comment (hiboabd) [inline spec/shared_contexts/with_a_logged_in_user.rb:6]: 👍

File: spec/system/casework/open_applications_dashboard_spec.rb [hunk 7]
```diff
@@ -68,55 +74,70 @@
...
-
     it 'includes tabs for work streams' do
       tabs = find(:xpath, "//div[@class='govuk-tabs']")
 
       expect(tabs).to have_content 'CAT 1 CAT 2 Extradition'
     end
 
-    context 'when viewing closed applications by work stream' do
-      it 'searches for extradition closed applications' do
+    context 'when viewing open applications by work stream' do
```
Comment (hiboabd) [inline spec/system/casework/open_applications_dashboard_spec.rb:83]: 😅 good spot

File: config/locales/en/casework.yml (hunk 4 above)
Comment (timpeat) [reply config/locales/en/casework.yml:68]: I've left it in as it could be in some test event data on staging. can remove later.

Comment (hiboabd) [reply config/locales/en/casework.yml:68]: Good point!

**Pull request merged**